tsdb/chunkenc/histogram.go

```diff
@@ -806,8 +804,8 @@ func (a *HistogramAppender) AppendHistogram(prev *HistogramAppender, t int64, h
 				panic(err) // This should never happen for an empty histogram chunk.
 			}
 			happ := app.(*HistogramAppender)
-			if counterReset {
-				happ.setCounterResetHeader(CounterReset)
+			if counterResetHeader != NotCounterReset {
```

Comment:
Isn't it better to always directly set the counter reset header and remove the `if`?